daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/RuntimeData.scala

```diff
@@ -763,12 +769,179 @@ final class ElementRuntimeData(
 
 }
 
+/**
+ * Used when unparsing to indicate that an expected ERD could not be created.
+ *
+ * Subclasses of this are for specific reasons why.
+ *
+ * The purposes of this is to allow the InfosetInputter to actually construct an
+ * infoset event including a full infoset DIElement node, yet indicating that the
+ * event is invalid.
+ *
+ * This enables improved diagnostic behavior. For example, a StartElement event
+ * can be created for a name and namespace where that name + namespace are not
+ * expected, the resulting infoset event will contain a DIElement having an
+ * ErrorElementRuntimeData ERD. The unparser can then inspect the infoset event, and
+ * if it is expecting an EndElement event, it can issue a UnparseError that
+ * correctly identifies that an expected EndElement event was not received.
+ *
+ * In all cases, there is no recovering from these errors in the Unparser; hence,
+ * we don't really care if these bogus DIElement nodes having these Error ERDs are
+ * spliced into the infoset or not.
+ */
+sealed abstract class ErrorERD(local: String, namespaceURI: String)
```

Comment:
Is there a better way to do this so that I don't have to construct this fully bogus object?
There's no union types in Scala..... I'd have to make ERD a trait, and then have the real and error subclasses.